packages/flutter/lib/src/rendering/shifted_box.dart

```diff
@@ -607,19 +607,46 @@ class RenderUnconstrainedBox extends RenderAligningShiftedBox with DebugOverflow
   RenderUnconstrainedBox({
     @required AlignmentGeometry alignment,
     @required TextDirection textDirection,
+    this.constrainedAxis,
     RenderBox child,
   }) : assert(alignment != null),
       super.mixin(alignment, textDirection, child);
 
+  /// The axis to retain constraints on, if any.
+  ///
+  /// If not set, or set to null (the default), neither axis will retain its
+  /// constraints.  If set to [Axis.vertical], then vertical constraints will
+  /// be retained, and if set to [Axis.horizontal], then horizontal constraints
+  /// will be retained.
+  Axis constrainedAxis;
```

Comment:
this needs a setter/getter pair to trigger layout when it changes (see all the other render objects for the pattern to follow).

if you add a test that varies this value over time you'll see that without the getter/setter pair, the layout doesn't update when the widget changes its constrainedAxis.

Reply:
Good point!  Added.